### DISCUSSION TITLE
How bookmarks are displayed in spectrogram.(and more)

### DISCUSSION THREAD

**Discussion by ageras1** [AlexandreRouma/SDRPlusPlus#211] [Ideas]
If you have bookmarks close together especially with long names, one hides another. i really like the implementation of gqrx.
Also off point i really like the spectrogram of gqrx (probably because i use it exclusively until now) i don't know exactly why but i think it is the coloring and the thickness of the lines makes it easier for me and as someone said in the issues "[Feature] Adjust spectrogram's vertical range by dragging the axis". A friend of mine who uses sdr# exclusively does not really like gqrx so....
Either way sdrpp is a great software.

**Comment by AlexandreRouma** [comment]
Could you describe how bookmark are supposed to be displayed?
If two bookmarks are close together, the only choice is either hiding their name or show them one on top of another

**Reply by ageras1** [reply]
![bookmarks](https://user-images.githubusercontent.com/88296235/127960881-6a6fe348-d6f5-492c-a8ae-cdcb3a35033e.png)
This is from Gqrx. As you can see at the left side it looks ok, at the right side it is just a mess but if you zoom in it gets fine (in hf you usually are zoomed in).
![bookmarks_zoomed](https://user-images.githubusercontent.com/88296235/127961564-ccc51e32-9f45-4a20-af85-97848016a6b4.png)
In another software they use the same aproach as you did but with two rows.

**Comment by ageras1** [comment]
I tried to understand how it is implemented in gqrx but while it looks "compact" i failed. I can not think a clean way to do something similar.
Maybe put first bookmark on row 1 and keep track of the free space. Then if the second bookmark collides in row 1 put in row 2 and also keep track of free space for the second row. Third bookmark first check row 1 and if it fits put it there, else go down a row and check.... It gets complicated.
Using two rows for the bookmarks seems to be "easy" to implement but i think as far as we have bookmarks changing how they are look should be lower in the "todo" list. If some has the time and knowledge (which i lack) can look at the gqrx code maybe it is not that hard to implement in the end.
Anyway thanks for the great work and because i use linux exclusively i am very glad this project exists also in linux. Bye
